src/Nethermind/Nethermind.Consensus/Validators/BlockValidator.cs

```diff
@@ -277,7 +311,46 @@ private bool ValidateWithdrawals(Block block, IReleaseSpec spec, out string? err
         return true;
     }
 
-    private bool ValidateTransactions(Block block, IReleaseSpec spec, out string? errorMessage)
+    public bool ValidateDeposits(Block block, out string? error) =>
+        ValidateDeposits(block, _specProvider.GetSpec(block.Header), out error);
+
+    private bool ValidateDeposits(Block block, IReleaseSpec spec, out string? error)
+    {
+        if (spec.IsEip6110Enabled && block.Deposits is null)
+        {
+            error = $"Deposits cannot be null in block {block.Hash} when EIP-6110 activated.";
```

Comment:
Check `BlockErrorMessages` and `TxErrorMessages` follow the pattern for error messages there.